### ISSUE TITLE
Break changes with the v0.8 (brackets)

### ISSUE THREAD

**Issue by antoinentl** [fabi1cazenave/qwerty-lafayette#101]
I try the 0.8 version, and I discovered a change for typing the brackets `[]{}`:

- v0.6:

![Capture d’écran du 2023-01-24 15-54-24](https://user-images.githubusercontent.com/6869488/214410930-ee49931b-eaa5-485e-8fb8-44373bac71e0.png)

- v0.8 (or v0.8.1):

![Capture d’écran du 2023-01-24 15-54-15](https://user-images.githubusercontent.com/6869488/214410647-f179dd28-8f7e-421c-badf-847a1c8264d6.png)

For me it's something like a break changes.
Is it an intentional change? If so, is it possible to keep the old _way_ for the brackets?

**Comment by fabi1cazenave** [comment]
It is an intentional change, yes.

TL;DR: do you really prefer getting these brackets with a dead key rather than with AltGr+ASDF ?

Now the long story…

In the very early days, one of the project goals was to make the AltGr key optional. So there’s always been a way to get those brackets `{[]}` and `<>` signs without AltGr, even on an ANSI keyboard. I think getting `<>` on dead key + ZX was okay-ish, but to me, these brackets on a same-pinky combo has always seemed the most unnatural thing ever.

Lafayette’s main goal is to bring a _better_ QWERTY. Having `{[]}` on the home row (with AltGr + ASDF) is a big part of it; and the whole AltGr layer is designed to make things like `->`, -=>`, `</`, `/>`, `(0)``, [1]`, `("")` as fluid as possible… and I thought, _maybe_ many users have missed that because there was another way to get those brackets — which would be a sign of bad design (poor self-discoverability).

So with 0.8, I’m proposing to split the project in two parts:
- the classic [Lafayette](https://qwerty-lafayette.org) layout, requiring 3 dead keys and an AltGr key;
- the [Lafayette42](https://qwerty-lafayette.org/42) variant, requiring a single dead key and featuring an even better but _really optional_ AltGr layer. 

If neither layout suits your needs, I’d be interested in reading your feedback.

**Comment by fabi1cazenave** [comment]
Pinning this issue, hoping to see as many replies as possible on this.

**Comment by fabi1cazenave** [comment]
@antoinentl your feedback would be interesting here. ;-)

- Do you really prefer getting these brackets with a dead key rather than with AltGr+ASDF ?
- If this is the case, isn’t the Lafayette42 variant best suited for you than Lafayette v0.6 ?

I’m not against keeping the old behavior *per se*, but I’d like to make sure I understand the use case.

**Comment by fabi1cazenave** [comment]
Thinking out loud, would this be a better default view to introduce Qwerty-Lafayette ?

![image](https://user-images.githubusercontent.com/457778/215283120-b1031d52-2379-4f4e-a714-883e03fa1bae.png)

**Comment by antoinentl** [comment]
@fabi1cazenave Thanks a lot for the explanation and the historic description, very interesting and useful! The clarification help me to understand the original goals and my misunderstood of the tool. I think I was not using QWERTY Lafayette at its full power, and your explanation made me understand it. I will learn to type AltGr+ASDF, it's more concise :rocket: + :keyboard: Go ahead with the v0.8, and the Lafayette42 variant is a possible alternative if that new version doesn't suite with my needs. (And if I want I can always go back to the v0.6.)

By the way the default view just above seems better!

You can close that issue when you want.

**Comment by fabi1cazenave** [comment]
Thanks for your contribution, that’s really appreciated. :-)
